### DISCUSSION TITLE
index.js: Unknown option: .resolver.

### DISCUSSION THREAD

**Discussion by kartik-stan** [microsoft/rnx-kit#2301] [Q&A] [answered]
### What happened?

I am trying to bundling but I ran into the following error while bundling the app
```index.js: Unknown option: .resolver. Check out https://babeljs.io/docs/en/babel-core/#options for more information about options.```

I am using the following command 
```react-native rnx-bundle --entry-file index.js --platform android --bundle-output android/main.jsbundle --sourcemap-output android/main.jsbundle.map --assets-dest android```

### bable.config.js
```
const { makeMetroConfig } = require("@rnx-kit/metro-config");
const MetroSymlinksResolver = require("@rnx-kit/metro-resolver-symlinks");

module.exports = makeMetroConfig({
    resolver: {
      resolveRequest: MetroSymlinksResolver(),
    },
    presets: ['@rnx-kit/babel-preset-metro-react-native'],
    plugins: ['react-native-reanimated/plugin'],
  });
```

Also I wanted to know if it is possible to serve the bundles from a server instead of keeping them in the app build? If so, is there any documentation around it?

Thanks in advance 

### Affected Package

@rnx-kit/cli

### Version

^0.16.4

### Which platforms are you seeing this issue on?

- [X] Android
- [X] iOS
- [ ] macOS
- [ ] Windows

### System Information

```true
System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1
    Memory: 162.52 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.1 - ~/.nvm/versions/node/v16.17.1/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.1/bin/npm
    Watchman: 2023.03.13.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.0 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      Android NDK: 21.4.7075529
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9619390
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.6 => 0.70.6 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to Reproduce

react-native rnx-bundle --entry-file index.js --platform android --bundle-output android/main.jsbundle --sourcemap-output android/main.jsbundle.map --assets-dest android

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tido64** [comment, marked as answer]
Hi @kartik-stan, it looks like you've mixed `babel.config.js` and `metro.config.js` together. Looking at the config you posted, I'm guessing they should be split like this:

```js
// babel.config.js
module.exports = {
  presets: ["@rnx-kit/babel-preset-metro-react-native"],
  plugins: ['react-native-reanimated/plugin'],
};

// metro.config.js
const { makeMetroConfig } = require("@rnx-kit/metro-config");
const MetroSymlinksResolver = require("@rnx-kit/metro-resolver-symlinks");

module.exports = makeMetroConfig({
  resolver: {
    resolveRequest: MetroSymlinksResolver(),
  },
});
```

> Also I wanted to know if it is possible to serve the bundles from a server instead of keeping them in the app build? If so, is there any documentation around it?

You can use `react-native rnx-start` where you would normally run `react-native start`. We have some documentation here: https://microsoft.github.io/rnx-kit/docs/guides/bundling#creating-bundler-commands
